hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndex.java

```diff
@@ -97,13 +100,32 @@ public abstract O updateLocation(O writeStatuses, HoodieEngineContext context,
   @PublicAPIMethod(maturity = ApiMaturityLevel.STABLE)
   public abstract boolean isImplicitWithStorage();
 
+  /**
+   * An index might need customized partitioner other than general upsert and insert partitioner.
+   */
+  @PublicAPIMethod(maturity = ApiMaturityLevel.EVOLVING)
+  public Option<Partitioner> getCustomizedPartitioner(WorkloadProfile profile,
+      HoodieEngineContext context,
+      HoodieTable table,
+      HoodieWriteConfig writeConfig) {
+    return Option.empty();
+  }
+
+  /**
+   * If the `getCustomizedPartitioner` returns a partitioner, it has to be true.
+   */
+  @PublicAPIMethod(maturity = ApiMaturityLevel.EVOLVING)
+  public boolean needCustomizedPartitioner() {
```

Comment:
can we remove this method and replace the callers of `needCustomzedPartitioner` to `getCustomzedPartitioner`, and determine whether it returns Option.empty?

Reply:
Work profile cannot be accessed where we call the `needCustomizedPartitioner `